Assets/BossRoom/Scripts/Server/Game/Entity/PublishMessageOnLifeChange.cs

```diff
@@ -0,0 +1,57 @@
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Server
+{
+    /// <summary>
+    /// Server-only component which publishes a message once the LifeState changes.
+    /// </summary>
+    [RequireComponent(typeof(NetworkLifeState))]
+    public class PublishMessageOnLifeChange : NetworkBehaviour
```

Comment:
We need a whole component just to publish a message?

Reply:
the goal here was to use composition to decide where to publish on, instead of making ServerCharacter even bigger and adding if-elses or switch statements to decide if we want to publish it or not (since we only want to publish it on player characters and the boss). I took inspiration from RaiseEventOnLifeChange, which we used for the boss.